src/Core/p2/iMath3.cpp

```diff
@@ -1,16 +1,115 @@
+  float32 in_f31;
+  double dVar4;
+  float local_38;
+  float local_34 [2];
+  //undefined auStack24 [16];
+  //undefined auStack8 [8];
+    float32 auStack24 [16];
+  float32 auStack8 [8];
+  __psq_st0(auStack8,(int)((unsigned long long)in_f31 >> 0x20),in_GQR0);
+  __psq_st1(auStack8,(int)in_f31,in_GQR0);
+  __psq_st0(auStack24,(int)((unsigned long long)in_f30 >> 0x20),in_GQR0);
+  __psq_st1(auStack24,(int)in_f30,in_GQR0);
+  if ((r->flags & 0x400) == 0) {
+    r->min_t = *(float32 *)(unaff_r2 + -0x68c4);
+  }
+  if ((r->flags & 0x800) == 0) {
+    r->max_t = *(float32 *)(unaff_r2 + -0x68c0);
+  }
+  xVec3Sub((xVec3 *)(isct + 0x1c),(xVec3 *)r,(xVec3 *)o);
+  dVar2 = (double)xVec3Dot((xVec3 *)(isct + 0x1c),(xVec3 *)(isct + 0x1c));
+  dVar3 = (double)(float)(dVar2 - (double)(*(float *)(o + 0xc) * *(float *)(o + 0xc)));
+  dVar2 = (double)xVec3Dot((xVec3 *)(isct + 0x1c),(xVec3 *)(r + 0xc));
+  dVar4 = (double)(float)((double)*(float *)(unaff_r2 + -0x68b8) * dVar2);
+  dVar2 = (double)xVec3Dot((xVec3 *)(r + 0xc),(xVec3 *)(r + 0xc));
+  iVar1 = xMathSolveQuadratic((float)dVar2,(float)dVar4,(float)dVar3,local_34,&local_38);
+  if (iVar1 == 0) {
+    isct->penned = (float32)(unaff_r2 + -0x68c0);
+    isct->contained = (float32)(unaff_r2 + -0x68c0);
+  }
+  else {
+    if (iVar1 == 1) {
+      if ((local_34[0] < *(float *)(r + 0x18)) || (*(float *)(r + 0x1c) < local_34[0]))
+      {
+        *(float *)(isct + 0x28) = local_34[0];
+        *(float32 *)(isct + 4) = *(float32 *)(unaff_r2 + -0x68c0);
+        *(float32 *)(isct + 8) = *(float32 *)(unaff_r2 + -0x68c0);
+      }
+      else {
+        *(float *)(isct + 0x28) = local_34[0];
+        *(float32 *)(isct + 4) = *(float32 *)(unaff_r2 + -0x68b4);
+        *(float32 *)(isct + 8) = *(float32 *)(unaff_r2 + -0x68c0);
+      }
+    }
+    else {
+      if (*(float *)(r + 0x18) <= local_34[0]) {
+        if (*(float *)(r + 0x1c) < local_34[0]) {
+          *(float *)(isct + 0x28) = local_34[0];
+          *(float32 *)(isct + 4) = *(float32 *)(unaff_r2 + -0x68c0);
+          *(float32 *)(isct + 8) = *(float32 *)(unaff_r2 + -0x68c0);
+        }
+        else {
+          *(float *)(isct + 0x28) = local_34[0];
+          *(float32 *)(isct + 4) = *(float32 *)(unaff_r2 + -0x68b4);
+          *(float32 *)(isct + 8) = *(float32 *)(unaff_r2 + -0x68c0);
+        }
+      }
+      else {
+        if (*(float *)(r + 0x18) <= local_38) {
+          *(float *)(isct + 0x28) = local_38;
+          *(float32 *)(isct + 4) = *(float32 *)(unaff_r2 + -0x68b4);
+          *(float32 *)(isct + 8) = *(float32 *)(unaff_r2 + -0x68b4);
+        }
+        else {
+          *(float *)(isct + 0x28) = local_38;
+          *(float32 *)(isct + 4) = *(float32 *)(unaff_r2 + -0x68c0);
+          *(float32 *)(isct + 8) = *(float32 *)(unaff_r2 + -0x68c0);
+        }
+      }
+    }
+  }
+  __psq_l0(auStack8,in_GQR0);
+  __psq_l1(auStack8,in_GQR0);
+  __psq_l0(auStack24,in_GQR0);
+  __psq_l1(auStack24,in_GQR0);
+  return;
+  #endif
```

Comment:
I get that this  probably is close to matching, but it's pretty useless as far as readable code goes. If you compare this function to `iSphereIsectVec`, it is much more legible and correct than this. This looks like you basically copied and pasted ghidra's output into a comment, which doesn't really help much honestly. If it almost matches and you want to keep it, this function needs to be cleaned up A LOT. Use the struct definitions to set the properties of a type, not random pointer math everywhere. Basically make this look like `iSphereIsectVec`, and use our custom types everywhere, and format the code with clang-format before I can accept this. Alternatively, you can submit the two other functions in a separate PR and I can merge them (after other issues are addressed), but I don't want to merge this code in

Reply:
Yea I was on 0 sleep when I made this PR. Tbh i shouldn't have made this PR yet. How about I just delete this PR until I have my bearing strait?